include/libraster.h

```diff
@@ -1,221 +1,161 @@
+ * \param[in] w Width of the rectangle
+ * \param[in] h Height of the rectangle
+ * \param[in] color Color of the rectangle (ARGB format)
  */
-typedef struct _Coords {
-    uint16_t x;
-    uint16_t y;
-} Coords;
-
-/**
-  * @brief Callback to draw a rectangle (may use DMA2D)
-  *
-  * @param x Position in pixel on x axis
-  * @param y Position in pixel on y axis
-  * @param w Width in pixel of rectangle
-  * @param h Height in pixel of rectangle
-  * @param color Color in ARGB8888 format
-  */
 typedef void (*draw_rectangle_callback_t)(uint16_t x, uint16_t y, uint16_t w, uint16_t h, uint32_t color);
 
-/**
-  * @brief Function used to clear screen (user decides the color)
-  */
+/*!
+ * \brief Function used to clear the screen
+ * 
+ * \details This callback function is used to clear the whole screen
+ */
 typedef void (*clear_screen_callback_t)(void);
 
-/**
- * @brief Defines one component of the UI
- *
- * @details
- *     - text Is a pointer to the string to plot
- *     - pos Is the position where to plot the text
- *     - font Font name (defined in font.h)
- *     - font_size Defines the size of the text
- *     - align Defines the alignement of the text
+/*!
+ * \brief Defines a label to be drawn on screen
  */
-typedef struct _Label {
-    char *text;
-    Coords pos;
-    FontName font;
-    uint16_t font_size;
-    FontAlign align;
-} Label;
-
-/**
- * @brief Defines a range for whom the color may be applied
- *
- * @details
- *     - min Is the min value for whom the color is applied
- *     - max Is the max value for whom the color is applied
- *     - bg_color Is the background color to apply
- *     - fg_color Is the foreground color to apply
+struct Label {
+    char *text;           /*!< Text to display */
+    struct Coords pos;    /*!< Position to draw the label */
+    enum FontName font;   /*!< Font name, defined in font.h */
+    uint16_t font_size;   /*!< Size of the text */
+    enum FontAlign align; /*!< Alignement of the text */
+};
+
+/*!
+ * \brief Defines one threshold for coloring
  */
-typedef struct _Threshold {
-    float min;
-    float max;
-    uint32_t bg_color;
-    uint32_t fg_color;
-} Threshold;
-
-/**
- * @brief Includes all the thresholds in one struct
- *
- * @details
- *     - threshold Pointer to Threshold struct
- *     - thresholds_num Number of thresholds in the array
+struct Threshold {
+    float min;         /*!< Minimum value of the threshold */
+    float max;         /*!< Maximum value of the threshold */
+    uint32_t bg_color; /*!< Background color to apply if in range */
+    uint32_t fg_color; /*!< Foreground color to apply if in range */
+};
+
+/*!
+ * \brief Defines multiple thresholds for coloring
  */
-typedef struct _Thresholds {
-    Threshold *threshold;
-    uint8_t thresholds_num;
-} Thresholds;
+struct Thresholds {
+    struct Threshold *threshold; /*!< Pointer to an array of thresholds */
+    uint8_t thresholds_num;      /*!< Number of thresholds present in the array */
```

Comment:
Since the field are defined inside the struct itself you can use simpler names, it does not work the same for enums for example since their fields are globally defined.
```suggestion
    uint8_t size;      /*!< Number of thresholds present in the array */
```